bug-1676470-pref-warpbuilder-experiment-release-83-83.toml

```diff
@@ -0,0 +1,369 @@
+
+        [metrics.load_input_event_response_ms.statistics.empirical_cdf]
+        pre_treatments = ["remove_nulls"]
+        log_space = true
+
+    [metrics.load_input_event_response_no_preload_ms]
+    select_expression = '{{agg_histogram_mean("payload.histograms.load_input_event_response_no_preload_ms")}}'
+    data_source = 'main'
+
+        [metrics.load_input_event_response_no_preload_ms.statistics.bootstrap_mean]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_no_preload_ms.statistics.deciles]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_no_preload_ms.statistics.kernel_density_estimate]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_no_preload_ms.statistics.empirical_cdf]
+        pre_treatments = ["remove_nulls"]
+        log_space = true
+
+    [metrics.load_input_event_response_preload_ms]
+    select_expression = '{{agg_histogram_mean("payload.histograms.load_input_event_response_preload_ms")}}'
+    data_source = 'main'
+
+        [metrics.load_input_event_response_preload_ms.statistics.bootstrap_mean]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_preload_ms.statistics.deciles]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_preload_ms.statistics.kernel_density_estimate]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.load_input_event_response_preload_ms.statistics.empirical_cdf]
+        pre_treatments = ["remove_nulls"]
+        log_space = true
+
+    [metrics.about_home_topsites_first_paint]
+    select_expression = 'SUM(COALESCE(payload.processes.parent.scalars.timestamps_about_home_topsites_first_paint))'
+    data_source = 'main'
+
+        [metrics.about_home_topsites_first_paint.statistics.bootstrap_mean]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.about_home_topsites_first_paint.statistics.deciles]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.about_home_topsites_first_paint.statistics.kernel_density_estimate]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.about_home_topsites_first_paint.statistics.empirical_cdf]
+        pre_treatments = ["remove_nulls"]
+        log_space = true
+
+    [metrics.fx_new_window_ms]
+    select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.fx_new_window_ms")}}'
+    data_source = 'main'
+
+        [metrics.fx_new_window_ms.statistics.bootstrap_mean]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.fx_new_window_ms.statistics.deciles]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.fx_new_window_ms.statistics.kernel_density_estimate]
+        pre_treatments = ["remove_nulls"]
+
+        [metrics.fx_new_window_ms.statistics.empirical_cdf]
+        pre_treatments = ["remove_nulls"]
+        log_space = true
+
+
+
+## Segments
```

Comment:
cpu_count and memory_mb are in clients_last_seen so you don't need to build a query from telemetry.main.

I think you _do_ need to specify some aggregation function over cpu_count and memory_mb, though, and you need to handle the case where there's no row in clients_last_seen -- so I think these would generally look something like `COALESCE(MAX(cpu_count) BETWEEN 1 AND 2, FALSE)`

Reply:
I can add it, but I don't understand why it would be needed? `clients_last_seen` already has client-day aggregation, so `MAX` shouldn't be needed (I should have put it in the custom segment using `main`). Same with `COALESCE`, there shouldn't be a case where a client is in the experiment and not in `clients_last_seen`.

Reply:
That's true, but BigQuery doesn't know those constraints, and will complain that we're only grouping by client_id and therefore need to specify an aggregation function for everything else we touch.

It's theoretically possible for a client to show up at 11:59 UTC and send an enrollment event before they ever send a main ping and make it into cls, and nulls crash the analysis, so the COALESCE provides a safety factor. Jetstream should probably choose to emit some warnings instead of crashing.

Reply:
(I'm unnerved that BigQuery is not already complaining; sigh)